cache/src/main/scala/org/allenai/common/cache/JsonQueryCacheModule.scala

```diff
@@ -4,14 +4,25 @@ import org.allenai.common.guice.ConfigModule
 
 import com.google.inject.Provides
 import com.google.inject.name.Named
-import com.typesafe.config.{ Config, ConfigFactory }
+import com.typesafe.config.Config
 import redis.clients.jedis.{ JedisPool, JedisPoolConfig }
+import spray.json.JsonFormat
+
+/** A guice module for binding a `JsonQueryCache` instance with a parameterized value type. */
+class JsonQueryCacheModule[V](config: Config)(
+  implicit
+  jsonFormat: JsonFormat[V],
+  manifest: Manifest[V]
```

Comment:
I think there was something like `ClassTag` that was preferred at one point or another.